src/midi/midi_fluidsynth.cpp

```diff
@@ -195,7 +195,16 @@ static std::vector<std_fs::path> get_data_dirs()
 		// of the actual path on Linux & Windows, so we need to
 		// normalise that to avoid some subtle bugs downstream (see
 		// `find_sf_file()` as well).
-		dirs.insert(dirs.begin(), std_fs::canonical(sf_dir));
+		if (path_exists(sf_dir)) {
+			dirs.insert(dirs.begin(), std_fs::canonical(sf_dir));
```

Comment:
I would consider using the form of `canonical` that returns an error code and never throws an exception:  https://en.cppreference.com/w/cpp/filesystem/canonical

Up to you.  I'll go ahead and approve either way.  Your way fixes the crash but I think it's more robust to prefer functions that don't throw exceptions as we really don't handle them properly.

Reply:
Good point, forgot about that. Changed, plus also fixed up `cdrom_ioctl_linux.cpp`.